File: 2581-divide-players-into-teams-of-equal-skill/2581-divide-players-into-teams-of-equal-skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        vector<pair<int,int>> ans;
        sort(skill.begin(),skill.end());
        if(skill.size()==2){
          return skill[0]*skill[1];
        }
        int val = skill[0]+skill[skill.size()-1];
        long long prod=0;
        for(int i=0;i<skill.size()/2;i++){
          if(val != (skill[i]+skill[skill.size()-i-1])){
            return -1;
          }
          prod += (long long)skill[i]*(long long)skill[skill.size()-i-1];
        }
        
        return prod;
    }
};

File: 2581-divide-players-into-teams-of-equal-skill/2581-divide-players-into-teams-of-equal-skill.py
class Solution(object):
    def dividePlayers(self, skill):
        """
        :type skill: List[int]
        :rtype: int
        """
        n = len(skill)
        skill.sort()

        if(n==2): return skill[0] * skill[1]

        val = skill[0] + skill[n-1]
        ans = 0

        for i in range(n//2):
            if val != (skill[i] + skill[-i-1]):
                return -1
            ans += (skill[i] * skill[-i-1])
        
        return ans


        